src/concurrency/transaction.cpp

```diff
@@ -172,7 +165,8 @@ bool Transaction::RecordDelete(const ItemPointer &location) {
     }
     PL_ASSERT(false);
   } else {
-    PL_ASSERT(false);
+    rw_set_[tile_group_id][tuple_id] = RWType::UPDATE;
```

Comment:
good catch here. should change `RWType::UPDATE` to `RWType::DELETE`. Similar as the previous one, a DBMS can delete a tuple after it `select_for_update` this tuple.